### PR TITLE
Fix transformer for blocks without name or position

### LINKED ISSUE #1496: Check why setPosition is set as null
I got this on branch `4.x`

```
Sonata\BlockBundle\Model\BaseBlock::setPosition(): Argument #1 ($position) must be of type int, null given, called in /app/vendor/sonata-project/page-bundle/src/Entity/Transformer.php on line 157
```

Stack trace

```
TypeError:
Sonata\BlockBundle\Model\BaseBlock::setPosition(): Argument #1 ($position) must be of type int, null given, called in /app/vendor/sonata-project/page-bundle/src/Entity/Transformer.php on line 157

  at vendor/sonata-project/block-bundle/src/Model/BaseBlock.php:147
  at Sonata\BlockBundle\Model\BaseBlock->setPosition(null)
     (vendor/sonata-project/page-bundle/src/Entity/Transformer.php:157)
  at Sonata\PageBundle\Entity\Transformer->loadBlock(array('id' => 11, 'name' => 'container 1', 'enabled' => true, 'position' => null, 'settings' => array('code' => null, 'layout' => '{{ CONTENT }}', 'class' => null, 'template' => '@SonataPage/Block/block_container.html.twig'), 'type' => 'sonata.block.service.container', 'created_at' => '1656522317', 'updated_at' => '1656522349', 'blocks' => array(array('id' => 9, 'name' => 'Test A', 'enabled' => true, 'position' => 1, 'settings' => array('mode' => 'form.choice_public', 'title' => 'Title B', 'translation_domain' => null, 'icon' => 'fa fa-globe', 'class' => null, 'template' => '@SonataPage/Block/block_pagelist.html.twig'), 'type' => 'sonata.page.block.pagelist', 'created_at' => '1656521962', 'updated_at' => '1656522564', 'blocks' => array()), array('id' => 10, 'name' => 'test s', 'enabled' => true, 'position' => 2, 'settings' => array('content' => 'Insert your custom content here', 'template' => '@SonataBlock/Block/block_core_text.html.twig'), 'type' => 'sonata.block.service.text', 'created_at' => '1656522079', 'updated_at' => '1656522388', 'blocks' => array()))), object(SnapshotPageProxy))
     (vendor/sonata-project/page-bundle/src/Model/SnapshotPageProxy.php:129)
  at Sonata\PageBundle\Model\SnapshotPageProxy->getBlocks()
     (vendor/sonata-project/page-bundle/src/CmsManager/CmsSnapshotManager.php:152)
  at Sonata\PageBundle\CmsManager\CmsSnapshotManager->loadBlocks(object(SnapshotPageProxy))
     (vendor/sonata-project/page-bundle/src/CmsManager/CmsSnapshotManager.php:133)
  at Sonata\PageBundle\CmsManager\CmsSnapshotManager->getPageBy(object(SonataPageSite), 'url', '/')
     (vendor/sonata-project/page-bundle/src/CmsManager/BaseCmsPageManager.php:54)
  at Sonata\PageBundle\CmsManager\BaseCmsPageManager->getPageByUrl(object(SonataPageSite), '/')
     (vendor/sonata-project/page-bundle/src/Route/CmsPageRouter.php:132)
  at Sonata\PageBundle\Route\CmsPageRouter->match('/')
     (vendor/symfony-cmf/routing/src/ChainRouter.php:192)
  at Symfony\Cmf\Component\Routing\ChainRouter->doMatch('/', object(Request))
     (vendor/symfony-cmf/routing/src/ChainRouter.php:158)
  at Symfony\Cmf\Component\Routing\ChainRouter->matchRequest(object(Request))
     (vendor/symfony/http-kernel/EventListener/RouterListener.php:112)
  at Symfony\Component\HttpKernel\EventListener\RouterListener->onKernelRequest(object(RequestEvent), 'kernel.request', object(TraceableEventDispatcher))
     (vendor/symfony/event-dispatcher/Debug/WrappedListener.php:126)
  at Symfony\Component\EventDispatcher\Debug\WrappedListener->__invoke(object(RequestEvent), 'kernel.request', object(TraceableEventDispatcher))
     (vendor/symfony/event-dispatcher/EventDispatcher.php:264)
  at Symfony\Component\EventDispatcher\EventDispatcher->doDispatch(array(object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener)), 'kernel.request', object(RequestEvent))
     (vendor/symfony/event-dispatcher/EventDispatcher.php:239)
  at Symfony\Component\EventDispatcher\EventDispatcher->callListeners(array(object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener)), 'kernel.request', object(RequestEvent))
     (vendor/symfony/event-dispatcher/EventDispatcher.php:73)
  at Symfony\Component\EventDispatcher\EventDispatcher->dispatch(object(RequestEvent), 'kernel.request')
     (vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcher.php:168)
  at Symfony\Component\EventDispatcher\Debug\TraceableEventDispatcher->dispatch(object(RequestEvent), 'kernel.request')
     (vendor/symfony/http-kernel/HttpKernel.php:134)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (vendor/symfony/http-kernel/HttpKernel.php:80)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (vendor/symfony/http-kernel/Kernel.php:201)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (public/index.php:29)   
```

But all my pages there are position value

Screenshot 2022-07-26 at 13 10 09

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because this only happens on 4.x branch.

Closes #1496

### CODE REVIEW

**Comment by eerison** [comment]
Hey @jordisala1991 I guess it wil work, but just for new register, should we add some message in the admin like "you are using a old version of snapshot, please generate a new one or something like this"

Because locally the devs doesn't use the data from prod and it won't show for them if we just trigger a normal deprecation. Make sense?

**Comment by eerison** [comment]
And without block name, I guess we should force to add name, we could use the same approach, if it's null ask to generate new snapshots

**Comment by eerison** [comment]
But what's happen if call getName or getPositon?

Can I return null?

**Comment by jordisala1991** [comment]
It should be able to return null, yes. But we need to see if it breaks on another point or it works.

**Comment by eerison** [comment]
> It should be able to return null, yes. But we need to see if it breaks on another point or it works.

In case someone uses really old snapshot with "title", yes it will break, because you are removing this workaround, but the dev just need to generate a new snapshot and it'll be fixed, but they need to be notified somehow that a new snapshot is required

**Comment by eerison** [comment]
Other thing what do you think we do not add the position property, instead of check if it's null or not 👀

File: src/Entity/Transformer.php [hunk 1]
```diff
@@ -216,30 +224,6 @@ public function getChildren(PageInterface $page)
         return $this->children[$page->getId()];
     }
 
-    /**
-     * @return array
-     */
-    private function fixPageContent(array $content)
-    {
-        if (!\array_key_exists('title', $content)) {
```
Comment (eerison) [inline src/Entity/Transformer.php:224]: Did you check if we are using title in other place, or it was just forget here?

Comment (jordisala1991) [reply src/Entity/Transformer.php:224]: I dont see the point on those fix content methods. the isset method already makes sure it exists and it is not null.

**Comment by jordisala1991** [comment]
> snapshot

I don't understand what you mean. Snapshot content without title will not do $page->setTitle, and snapshot with title will call $page->setTitle, or am I missing something?

File: src/Entity/Transformer.php [hunk 2]
```diff
@@ -119,13 +119,17 @@ public function load(SnapshotInterface $snapshot)
...
-        $content = $this->fixPageContent($snapshot->getContent());
+        $content = $snapshot->getContent();
 
         $page->setId($content['id']);
         $page->setJavascript($content['javascript']);
         $page->setStylesheet($content['stylesheet']);
         $page->setRawHeaders($content['raw_headers']);
-        $page->setTitle($content['title']);
+
+        if (isset($content['title'])) {
```
Comment (eerison) [inline src/Entity/Transformer.php:129]: Yeah i guess it'll check, if it's exist and it's not null

Then it's fine

**Comment by jordisala1991** [comment]
> Other thing what do you think we do not add the position property, instead of check if it's null or not 👀

Doing an isset should already check for null.

File: src/Entity/Transformer.php (hunk 2 above)
Comment (jordisala1991) [reply src/Entity/Transformer.php:129]: Yes! https://3v4l.org/b8qF0

**Comment by eerison** [comment]
I guess it's fine for now, but we should create a other ticket to solve this mess 😢

**Comment by eerison** [comment]
For me it's fine, I guess it'll stop to crash the page :), after merge, I can test again ;)

**Comment by jordisala1991** [comment]
> I guess it's fine for now, but we should create a other ticket to solve this mess 😢

We should probably implement another object with the responsability to check wheter a json stored on the db is a valid block / page structure or is outdated/invalid.

TBH I never used this bundle to know exactly at what point we should inform the user that its snapshot contents are kind of outdated or invalid.

I am not sure yet about how to properly solve this. I guess increasing PHPStan will start giving hints on where this use of arrays is problematic.

What I am sure is working with arrays that comes from the user database is a bit of a mess, this transformer can't assume much about what is stored on the db. Other properties might need to be wrapped on a isset call later.

**Comment by jordisala1991** [comment]
> For me it's fine, I guess it'll stop to crash the page :), after merge, I can test again ;)

You can test before the merge too. Just need to add my repository on composer json an use my branch as your sonatapagebundle version (this way you can test PRs before merge). Example:

https://getcomposer.org/doc/05-repositories.md#loading-a-package-from-a-vcs-repository

```json
{
    "repositories": [
        {
            "type": "vcs",
            "url": "https://github.com/jordisala1991/SonataPageBundle"
        }
    ],
    "require": {
        "sonata-project/page-bundle": "dev-hotfix/transformer as 4.0.0"
    }
}
```

**Comment by eerison** [comment]
> > For me it's fine, I guess it'll stop to crash the page :), after merge, I can test again ;)
> 
> You can test before the merge too. Just need to add my repository on composer json an use my branch as your sonatapagebundle version (this way you can test PRs before merge). Example:
> 
> https://getcomposer.org/doc/05-repositories.md#loading-a-package-from-a-vcs-repository
> 
> ```json
> {
>     "repositories": [
>         {
>             "type": "vcs",
>             "url": "https://github.com/jordisala1991/SonataPageBundle"
>         }
>     ],
>     "require": {
>         "sonata-project/page-bundle": "dev-hotfix/transformer as 4.0.0"
>     }
> }
> ```
I will check ;)

**Comment by eerison** [comment]
Hey @jordisala1991 I'm still getting a error, related with `setPosition`

Screenshot 2022-07-27 at 08 29 14


<details><summary>Stack trace</summary>
<p>

```
TypeError:
Sonata\BlockBundle\Model\BaseBlock::setPosition(): Argument #1 ($position) must be of type int, string given, called in /app/vendor/sonata-project/page-bundle/src/Entity/Transformer.php on line 165

  at vendor/sonata-project/block-bundle/src/Model/BaseBlock.php:147
  at Sonata\BlockBundle\Model\BaseBlock->setPosition('0')
     (vendor/sonata-project/page-bundle/src/Entity/Transformer.php:165)
  at Sonata\PageBundle\Entity\Transformer->loadBlock(array('id' => 25, 'name' => null, 'enabled' => true, 'position' => '0', 'settings' => array('url' => 'https://facebook.com', 'title' => 'RSS feed', 'translation_domain' => null, 'icon' => 'fa fa-rss-square', 'class' => null, 'template' => '@SonataBlock/Block/block_core_rss.html.twig'), 'type' => 'sonata.block.service.rss', 'created_at' => '1656574713', 'updated_at' => '1658828870', 'blocks' => array()), object(SnapshotPageProxy))
     (vendor/sonata-project/page-bundle/src/Entity/Transformer.php:180)
  at Sonata\PageBundle\Entity\Transformer->loadBlock(array('id' => 16, 'name' => 'Main content', 'enabled' => true, 'position' => 1, 'settings' => array('code' => 'content'), 'type' => 'sonata.page.block.container', 'created_at' => '1656571318', 'updated_at' => '1656571318', 'blocks' => array(array('id' => 25, 'name' => null, 'enabled' => true, 'position' => '0', 'settings' => array('url' => 'https://facebook.com', 'title' => 'RSS feed', 'translation_domain' => null, 'icon' => 'fa fa-rss-square', 'class' => null, 'template' => '@SonataBlock/Block/block_core_rss.html.twig'), 'type' => 'sonata.block.service.rss', 'created_at' => '1656574713', 'updated_at' => '1658828870', 'blocks' => array()), array('id' => 19, 'name' => null, 'enabled' => true, 'position' => '1', 'settings' => array('content' => 'SIMPLE TEXT :D', 'template' => '@SonataBlock/Block/block_core_text.html.twig'), 'type' => 'sonata.block.service.text', 'created_at' => '1656571374', 'updated_at' => '1658828870', 'blocks' => array()), array('id' => 23, 'name' => 'aaa', 'enabled' => true, 'position' => '2', 'settings' => array('title' => 'Breadcrumb (Homepage)', 'cache_policy' => 'public', 'template' => '@SonataBlock/Block/block_core_menu.html.twig', 'menu_name' => null, 'safe_labels' => false, 'current_class' => 'active', 'first_class' => null, 'last_class' => null, 'current_uri' => null, 'menu_class' => 'list-group', 'children_class' => 'list-group-item', 'menu_template' => '@SonataSeo/Block/breadcrumb.html.twig', 'include_homepage_link' => true, 'context' => false), 'type' => 'sonata.seo.block.breadcrumb.homepage', 'created_at' => '1656571781', 'updated_at' => '1658828870', 'blocks' => array()), array('id' => 24, 'name' => null, 'enabled' => true, 'position' => '3', 'settings' => array('url' => 'https://google.com', 'title' => 'google link', 'translation_domain' => null, 'icon' => 'fa fa-rss-square', 'class' => null, 'template' => '@SonataBlock/Block/block_core_rss.html.twig'), 'type' => 'sonata.block.service.rss', 'created_at' => '1656574614', 'updated_at' => '1658828870', 'blocks' => array()), array('id' => 26, 'name' => 'Instagram', 'enabled' => true, 'position' => '4', 'settings' => array('url' => 'https://instagram.com', 'title' => 'INSTA PROD 45678', 'translation_domain' => null, 'icon' => 'fa fa-rss-square', 'class' => null, 'template' => '@SonataBlock/Block/block_core_rss.html.twig'), 'type' => 'sonata.block.service.rss', 'created_at' => '1656574756', 'updated_at' => '1658828870', 'blocks' => array()), array('id' => 27, 'name' => null, 'enabled' => true, 'position' => '5', 'settings' => array('template' => '@SonataSeo/Block/block_email_share_button.html.twig', 'subject' => 'test 12455', 'body' => 'test 4333'), 'type' => 'sonata.seo.block.email.share_button', 'created_at' => '1657118800', 'updated_at' => '1658828870', 'blocks' => array()))), object(SnapshotPageProxy))
     (vendor/sonata-project/page-bundle/src/Model/SnapshotPageProxy.php:129)
  at Sonata\PageBundle\Model\SnapshotPageProxy->getBlocks()
     (vendor/sonata-project/page-bundle/src/CmsManager/CmsSnapshotManager.php:152)
  at Sonata\PageBundle\CmsManager\CmsSnapshotManager->loadBlocks(object(SnapshotPageProxy))
     (vendor/sonata-project/page-bundle/src/CmsManager/CmsSnapshotManager.php:133)
  at Sonata\PageBundle\CmsManager\CmsSnapshotManager->getPageBy(object(SonataPageSite), 'url', '/contact')
     (vendor/sonata-project/page-bundle/src/CmsManager/BaseCmsPageManager.php:54)
  at Sonata\PageBundle\CmsManager\BaseCmsPageManager->getPageByUrl(object(SonataPageSite), '/contact')
     (vendor/sonata-project/page-bundle/src/Route/CmsPageRouter.php:132)
  at Sonata\PageBundle\Route\CmsPageRouter->match('/contact')
     (vendor/symfony-cmf/routing/src/ChainRouter.php:192)
  at Symfony\Cmf\Component\Routing\ChainRouter->doMatch('/contact', object(Request))
     (vendor/symfony-cmf/routing/src/ChainRouter.php:158)
  at Symfony\Cmf\Component\Routing\ChainRouter->matchRequest(object(Request))
     (vendor/symfony/http-kernel/EventListener/RouterListener.php:112)
  at Symfony\Component\HttpKernel\EventListener\RouterListener->onKernelRequest(object(RequestEvent), 'kernel.request', object(TraceableEventDispatcher))
     (vendor/symfony/event-dispatcher/Debug/WrappedListener.php:126)
  at Symfony\Component\EventDispatcher\Debug\WrappedListener->__invoke(object(RequestEvent), 'kernel.request', object(TraceableEventDispatcher))
     (vendor/symfony/event-dispatcher/EventDispatcher.php:264)
  at Symfony\Component\EventDispatcher\EventDispatcher->doDispatch(array(object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener)), 'kernel.request', object(RequestEvent))
     (vendor/symfony/event-dispatcher/EventDispatcher.php:239)
  at Symfony\Component\EventDispatcher\EventDispatcher->callListeners(array(object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener)), 'kernel.request', object(RequestEvent))
     (vendor/symfony/event-dispatcher/EventDispatcher.php:73)
  at Symfony\Component\EventDispatcher\EventDispatcher->dispatch(object(RequestEvent), 'kernel.request')
     (vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcher.php:168)
  at Symfony\Component\EventDispatcher\Debug\TraceableEventDispatcher->dispatch(object(RequestEvent), 'kernel.request')
     (vendor/symfony/http-kernel/HttpKernel.php:134)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (vendor/symfony/http-kernel/HttpKernel.php:80)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (vendor/symfony/http-kernel/Kernel.php:201)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (public/index.php:29)
```
</p>
</details>

Just other point: 
I guess it's checking all snapshots from my page, because the latest snapshot generated, there isn't `positions=null`


<details><summary>Content json</summary>
<p>

```json
{
   "id":652,
   "name":"contact",
   "javascript":null,
   "stylesheet":null,
   "raw_headers":null,
   "title":null,
   "meta_description":null,
   "meta_keyword":null,
   "template_code":"default",
   "request_method":"GET|POST|HEAD|DELETE|PUT",
   "created_at":"1656522709",
   "updated_at":"1657119067",
   "slug":"contact",
   "parent_id":642,
   "target_id":null,
   "blocks":[
      {
         "id":15,
         "name":"Top content",
         "enabled":true,
         "position":1,
         "settings":{
            "code":"content_top"
         },
         "type":"sonata.page.block.container",
         "created_at":"1656571318",
         "updated_at":"1656571318",
         "blocks":[
            {
               "id":21,
               "name":null,
               "enabled":true,
               "position":1,
               "settings":{
                  "template":"@SonataSeo\/Block\/block_email_share_button.html.twig",
                  "subject":"subject test",
                  "body":"body test"
               },
               "type":"sonata.seo.block.email.share_button",
               "created_at":"1656571654",
               "updated_at":"1656571654",
               "blocks":[
                  
               ]
            },
            {
               "id":22,
               "name":null,
               "enabled":true,
               "position":2,
               "settings":{
                  "mode":"form.choice_public",
                  "title":"ABBB",
                  "translation_domain":null,
                  "icon":"fa fa-globe",
                  "class":null,
                  "template":"@SonataPage\/Block\/block_pagelist.html.twig"
               },
               "type":"sonata.page.block.pagelist",
               "created_at":"1656571709",
               "updated_at":"1656571709",
               "blocks":[
                  
               ]
            }
         ]
      },
      {
         "id":16,
         "name":"Main content",
         "enabled":true,
         "position":1,
         "settings":{
            "code":"content"
         },
         "type":"sonata.page.block.container",
         "created_at":"1656571318",
         "updated_at":"1656571318",
         "blocks":[
            {
               "id":25,
               "name":null,
               "enabled":true,
               "position":0,
               "settings":{
                  "url":"https:\/\/facebook.com",
                  "title":"RSS feed",
                  "translation_domain":null,
                  "icon":"fa fa-rss-square",
                  "class":null,
                  "template":"@SonataBlock\/Block\/block_core_rss.html.twig"
               },
               "type":"sonata.block.service.rss",
               "created_at":"1656574713",
               "updated_at":"1657119037",
               "blocks":[
                  
               ]
            },
            {
               "id":27,
               "name":null,
               "enabled":true,
               "position":1,
               "settings":{
                  "template":"@SonataSeo\/Block\/block_email_share_button.html.twig",
                  "subject":"test 12455",
                  "body":"test 4333"
               },
               "type":"sonata.seo.block.email.share_button",
               "created_at":"1657118800",
               "updated_at":"1657119037",
               "blocks":[
                  
               ]
            },
            {
               "id":19,
               "name":null,
               "enabled":true,
               "position":2,
               "settings":{
                  "content":"SIMPLE TEXT :D",
                  "template":"@SonataBlock\/Block\/block_core_text.html.twig"
               },
               "type":"sonata.block.service.text",
               "created_at":"1656571374",
               "updated_at":"1657119037",
               "blocks":[
                  
               ]
            },
            {
               "id":23,
               "name":"aaa",
               "enabled":true,
               "position":3,
               "settings":{
                  "title":"Breadcrumb (Homepage)",
                  "cache_policy":"public",
                  "template":"@SonataBlock\/Block\/block_core_menu.html.twig",
                  "menu_name":null,
                  "safe_labels":false,
                  "current_class":"active",
                  "first_class":null,
                  "last_class":null,
                  "current_uri":null,
                  "menu_class":"list-group",
                  "children_class":"list-group-item",
                  "menu_template":"@SonataSeo\/Block\/breadcrumb.html.twig",
                  "include_homepage_link":true,
                  "context":false
               },
               "type":"sonata.seo.block.breadcrumb.homepage",
               "created_at":"1656571781",
               "updated_at":"1657119037",
               "blocks":[
                  
               ]
            },
            {
               "id":24,
               "name":null,
               "enabled":true,
               "position":4,
               "settings":{
                  "url":"https:\/\/google.com",
                  "title":"google link",
                  "translation_domain":null,
                  "icon":"fa fa-rss-square",
                  "class":null,
                  "template":"@SonataBlock\/Block\/block_core_rss.html.twig"
               },
               "type":"sonata.block.service.rss",
               "created_at":"1656574614",
               "updated_at":"1657119037",
               "blocks":[
                  
               ]
            },
            {
               "id":26,
               "name":"Instagram",
               "enabled":true,
               "position":5,
               "settings":{
                  "url":"https:\/\/instagram.com",
                  "title":"INSTA PROD 45678",
                  "translation_domain":null,
                  "icon":"fa fa-rss-square",
                  "class":null,
                  "template":"@SonataBlock\/Block\/block_core_rss.html.twig"
               },
               "type":"sonata.block.service.rss",
               "created_at":"1656574756",
               "updated_at":"1657119037",
               "blocks":[
                  
               ]
            }
         ]
      },
      {
         "id":17,
         "name":"Bottom content",
         "enabled":true,
         "position":1,
         "settings":{
            "code":"content_bottom"
         },
         "type":"sonata.page.block.container",
         "created_at":"1656571318",
         "updated_at":"1656571318",
         "blocks":[
            
         ]
      },
      {
         "id":18,
         "name":"Footer",
         "enabled":true,
         "position":1,
         "settings":{
            "code":"footer"
         },
         "type":"sonata.page.block.container",
         "created_at":"1656571318",
         "updated_at":"1656571318",
         "blocks":[
            
         ]
      },
      {
         "id":14,
         "name":"Header",
         "enabled":true,
         "position":1,
         "settings":{
            "code":"header"
         },
         "type":"sonata.page.block.container",
         "created_at":"1656571318",
         "updated_at":"1656571318",
         "blocks":[
            {
               "id":20,
               "name":"Menu 1",
               "enabled":true,
               "position":1,
               "settings":{
                  "title":"sonata.block.menu",
                  "cache_policy":"public",
                  "template":"@SonataBlock\/Block\/block_core_menu.html.twig",
                  "menu_name":"sonata_admin_sidebar",
                  "safe_labels":false,
                  "current_class":"active",
                  "first_class":null,
                  "last_class":null,
                  "current_uri":null,
                  "menu_class":"list-group",
                  "children_class":"list-group-item",
                  "menu_template":null
               },
               "type":"sonata.block.service.menu",
               "created_at":"1656571563",
               "updated_at":"1656571563",
               "blocks":[
                  
               ]
            }
         ]
      }
   ]
}
```

</p>
</details>

**Comment by jordisala1991** [comment]
Nice, seems like you have position="0" as string 😓

**Comment by eerison** [comment]
> Nice, seems like you have position="0" as string 😓

Maybe it's converted to integer? and it's parsed to `0`? because I guess it's not saved with "0" (string) into the database

**Comment by eerison** [comment]
@jordisala1991 does make sense I investigate this issue too?, or should I do other one like this: #1497?

**Comment by jordisala1991** [comment]
I will need you to test this branch, when I do changes I will let you know

**Comment by jordisala1991** [comment]
Most of the PR is already on 4.x, the issue is not fixed tho, but it will need a new PR.

**Pull request closed without merging**